hack/e2e-common.sh

```diff
@@ -55,5 +55,6 @@ function cluster_kueue_deploy {
     else
         kubectl apply --server-side -k test/e2e/config
     fi
+    kubectl wait --for=condition=available --timeout=3m deployment/kueue-controller-manager -n kueue-system
```

Comment:
Can we do this somewhere else ?  

It could help if in case of multikue , we deploy in manager, and instead of waiting to be ready now, we also deploy in the workers , and check the availability later on for all of them.

Reply:
I see, this is a simple way of waiting for Kueue ready. I imagine users would like to use in their setup scripts. Typically it takes a couple of seconds.

One alternative is to revert the removal of  `KueueReadyForTesting`, but I don't like it, because it recreates objects, this takes time and could cover some issues (as it is actually currently in case of single cluster). Actually, in this PR I wanted to remove it to test that it is not necessary.

Other alternatives I see modify `KueueReadyForTesting` to check the deployment status as this line, or run the entire setup of different clusters for MK in parallel at the script level.

Reply:
+1 on changing `KueueReadyForTesting` with an availability check for the deployment.

Reply:
Let's leave this for a follow up.

Reply:
Sure, but I'm not sure what is the preferred option as a follow up. Is it returning to the `KueueReadyForTesting`, but waiting for deployment available, as in the command line?